src/xrpld/app/tx/detail/Batch.cpp

```diff
@@ -0,0 +1,251 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/tx/apply.h>
+#include <xrpld/app/tx/detail/Batch.h>
+#include <xrpld/ledger/Sandbox.h>
+#include <xrpld/ledger/View.h>
+#include <xrpl/basics/Log.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/Indexes.h>
+#include <xrpl/protocol/TER.h>
+#include <xrpl/protocol/TxFlags.h>
+
+namespace ripple {
+
+XRPAmount
+Batch::calculateBaseFee(ReadView const& view, STTx const& tx)
+{
+    // Calculate the Inner Txn Fees
+    XRPAmount txnFees{0};
+    if (tx.isFieldPresent(sfRawTransactions))
+    {
+        XRPAmount txFees{0};
+        auto const& txns = tx.getFieldArray(sfRawTransactions);
+        for (STObject txn : txns)
+        {
+            STTx const stx = STTx{std::move(txn)};
+            txFees += ripple::calculateBaseFee(view, stx);
+        }
+        txnFees += txFees;
+    }
+
+    // Calculate the BatchSigners Fees
+    std::int32_t signerCount = tx.isFieldPresent(sfBatchSigners)
+        ? tx.getFieldArray(sfBatchSigners).size()
+        : 0;
+
+    // sum of inner tx fees + 10 drops per signature + 20 drops for processing
+    return ((signerCount + 2) * view.fees().base) + txnFees;
```

Comment:
This equation doesn't seem to take into account `tx.Signers` and `tx.BatchSigners.Signers`